sdk/core/core-client-rest/src/sendRequest.ts

```diff
@@ -93,17 +99,32 @@ function getContentType(body: any): string | undefined {
   return "application/json; charset=UTF-8";
 }
 
-export interface InternalRequestParameters extends RequestParameters {
+export type InternalRequestParameters = RequestParameters & {
   responseAsStream?: boolean;
-}
+};
 
 function buildPipelineRequest(
   method: HttpMethods,
   url: string,
   options: InternalRequestParameters = {},
 ): PipelineRequest {
   const requestContentType = getRequestContentType(options);
-  const { body, formData } = getRequestBody(options.body, requestContentType);
+
+  let body: RequestBodyType | undefined;
+  let formData: FormDataMap | undefined;
+  let multipartBody: MultipartRequestBody | undefined;
+
+  switch (options.bodyKind) {
```

Comment:
where is the option defined? is it in InternalRequestParameters?

Reply:
It's defined in each of the types of the new union definition of `RequestParameters`